lib/catalog/data_driven_form_validator.rb

```diff
@@ -0,0 +1,86 @@
+module Catalog
+  class DataDrivenFormValidator
+    COMPONENTS = %i[text-field
+                    textarea-field
+                    field-array
+                    select-field
+                    checkbox
+                    sub-form
+                    radio
+                    tabs
+                    tab-item
+                    date-picker
+                    time-picker
+                    wizard
+                    switch-field
+                    plain-text].freeze
+
+    VALIDATORS = %i[required-validator
+                    min-length-validator
+                    max-length-validator
+                    exact-length-validator
+                    min-items-validator
+                    min-number-value
+                    max-number-value
+                    pattern-validator
+                    url-validator].freeze
+
+    DATA_TYPES = %i[integer
+                    float
+                    number
+                    boolean
+                    string].freeze
+
+    def self.valid?(ddf)
+      ddf = ddf.class == Hash ? ddf.with_indifferent_access : JSON.parse(ddf).with_indifferent_access
+      check_fields(ddf[:schema][:fields])
+
+      true
+    end
+
+    class << self
```

Comment:
Any reason this one isn't above line 34 with 34 changing to `def valid?(ddf)`? Or were these intended to be `private` and that keyword just needs to be here after the `class << self`?

Reply:
Yeah they're `private`, but the `private` keyword doesn't work for class methods apparently so I'd have to use `private_class_method` which screws up formatting, so I went with `class << self` instead.